Refactor runtime documentation and add hardened Docker installation guide (#7429)
All-Hands-AI/OpenHands: docs/modules/usage/runtimes/daytona.md

# Daytona Runtime

> Another option is using [Daytona](https://www.daytona.io/) as a runtime provider:

Reviewer: ```suggestion
You can use [Daytona](https://www.daytona.io/) as a runtime provider:
```
